x-pack/plugins/enterprise_search/public/applications/workplace_search/components/overview/overview_logic.ts

```diff
@@ -45,119 +51,102 @@ export interface IOverviewValues extends IOverviewServerData {
   flashMessages: IFlashMessagesProps;
 }
 
-interface IListenerParams {
-  actions: IOverviewActions;
-}
-
-interface IOverviewLogic extends IKeaLogic, IListenerParams {
-  values: IOverviewValues;
-}
-
 export const OverviewLogic = kea({
   actions: (): IOverviewActions => ({
-    setServerData: (serverData: IOverviewServerData) => serverData,
-    setFlashMessages: (flashMessages: IFlashMessagesProps) => ({
-      flashMessages,
-    }),
-    setHasErrorConnecting: (hasErrorConnecting: boolean) => ({ hasErrorConnecting }),
-    initializeOverview: ({ http }: { http: HttpSetup }) => ({ http }),
+    setServerData: (serverData) => serverData,
+    setFlashMessages: (flashMessages) => ({ flashMessages }),
+    setHasErrorConnecting: (hasErrorConnecting) => ({ hasErrorConnecting }),
+    initializeOverview: ({ http }) => ({ http }),
   }),
-  reducers: () => ({
+  reducers: (): TKeaReducers<IOverviewValues, IOverviewActions> => ({
     organization: [
-      {},
+      {} as IOrganization,
```

Comment:
You can test out the new default state type checking by changing this to something like

```suggestion
      'invalid',
```

And you should see in VSCode that it errors with `Type 'string' is not assignable to type 'IOrganization'.`